crates/biome_formatter/src/trivia.rs

```diff
@@ -245,21 +308,30 @@ where
         if dangling_comments.is_empty() {
             return Ok(());
         }
-
+        // Write all comments up to the first skipped token trivia or the token
         let format_dangling_comments = format_with(|f| {
-            // Write all comments up to the first skipped token trivia or the token
-            let mut join = f.join_with(hard_line_break());
```

Comment:
General question: are `JoinBuilder`s particularly efficient for building here? Or just convenient? This _can't_ use a joiner anymore since the joining element can be different based on the context, but idk if that has a performance hit or not.

The convenience is nice to not have to implement the awkward conditional below on Line 326 lol, but oh well.